pack.go

```diff
@@ -43,38 +44,85 @@ const (
 	MediaTypeUnknownArtifact = "application/vnd.unknown.artifact.v1"
 )
 
-// PackManifestType represents the manifest type used for [Pack].
-type PackManifestType int
-
-const (
-	// PackManifestTypeImageV1_1_0_RC2 represents the OCI Image Manifest type
-	// defined in image-spec v1.1.0-rc2.
-	// Reference: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/manifest.md
-	//
-	// Deprecated: This type is deprecated and not recommended for future use.
-	// Use PackManifestTypeImageV1_1_0_RC4 instead.
-	PackManifestTypeImageV1_1_0_RC2 PackManifestType = 0
-
-	// PackManifestTypeImageV1_1_0_RC4 represents the OCI Image Manifest type
-	// defined since image-spec v1.1.0-rc4.
-	// Reference: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc4/manifest.md
-	PackManifestTypeImageV1_1_0_RC4 PackManifestType = 1
-)
-
 var (
-	// ErrInvalidDateTimeFormat is returned by [Pack] when
+	// ErrInvalidDateTimeFormat is returned by [Pack] and [PackManifest] when
 	// AnnotationArtifactCreated or AnnotationCreated is provided, but its value
 	// is not in RFC 3339 format.
 	// Reference: https://www.rfc-editor.org/rfc/rfc3339#section-5.6
 	ErrInvalidDateTimeFormat = errors.New("invalid date and time format")
 
-	// ErrMissingArtifactType is returned by [Pack] when artifactType is not
-	// specified and the config media type is set to
+	// ErrMissingArtifactType is returned by [PackManifest] when
+	// packManifestType is PackManifestVersion1_1_RC4 and artifactType is
+	// empty and the config media type is set to
 	// "application/vnd.oci.empty.v1+json".
 	ErrMissingArtifactType = errors.New("missing artifact type")
 )
 
+// PackManifestVersion represents the manifest version used for [PackManifest].
+type PackManifestVersion int
+
+const (
+	// PackManifestVersion1_0 represents the OCI Image Manifest defined in
+	// image-spec v1.0.2.
+	// Reference: https://github.com/opencontainers/image-spec/blob/v1.0.2/manifest.md
+	PackManifestVersion1_0 PackManifestVersion = 1
+
+	// PackManifestVersion1_1_RC4 represents the OCI Image Manifest defined
+	// in image-spec v1.1.0-rc4.
+	// Reference: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc4/manifest.md
+	PackManifestVersion1_1_RC4 PackManifestVersion = 2
+)
+
+// PackManifestOptions contains parameters for [PackManifest].
+type PackManifestOptions struct {
```

Comment:
Why `artifactType` is not put in this struct?

Reply:
Two reasons:
1. We encourage users to set this value
2. This value must be set when config is empty

Reply:
`PackManifestOptions` contains optional parameters only.

Reply:
Maybe we should mention `optional` in the doc.